### ISSUE TITLE
How does fine tuning work? AND wonder suitable loss

### ISSUE THREAD

**Issue by h19920918** [thtrieu/darkflow#258]
Thank you for your contribution.

I try to train own dataset and my training set has one more task which is estimation pose of object.
I finish to modify internal codes, and training is working.

When I use pre-trained weights, Pre-trained weights model and my new training model have different shape for only output layer.
In my guessing, it doesn't work. But it works. So, I wonder how to operate it.

Second, What is suitable loss when we train? My average training loss doesn't drop under 2.5.

**Comment by Xianlei** [comment]
Same problem here.

If we want to train on new data set from ground, we need to change the **classes** in [detection] layer and **output** in last [connect] layer. 

What **else** should we change?

Also, how to do a transfer learning from a pre-trained weight? since we already change last connect layer, we need to avoid loading weights for that layer.

thanks.

@thtrieu

**Comment by Xianlei** [comment]
Besides, I also got training loss around 2.5.

**Comment by h19920918** [comment]
I make training loss dropped until 1.00~ now.
First, I train own my dataset with pre-trained weights using learning rate 1e-4 for 20 epoch.
After finish training, change lr 1e-5 for 40 epoch and train now using lr 1e-6.

But I don't know suitable loss.

I try to train own dataset without pre-trained weights.
Starting learning rate is 1e-3.

**Comment by Xianlei** [comment]
@h19920918  how does it perform on prediction ? does it behave right?

I saw in another post the loss should be around 0.

**Comment by h19920918** [comment]
Yes, I changed from loading xml to output cython code.

But prediction result is too bad.
If you know mAVP, the result is 0.25~ and recall point could not be 1.
So, my measurement is something wrong that I think.

mAVP, mAP code was made on my hand.

**Comment by Xianlei** [comment]
Which model did you use by  the way? My prediction is also bad..... 

I only trained on 480 images and tried to overfit this. but always get around 2.0 loss

**Comment by h19920918** [comment]
I don't know what you say about model.
I used variation of yolo-voc.cfg for fitting own dataset.

The number of my training instances are 21165 images.
Now, my training loss is nearby 0.95xx~.

I recommend you need to get more data.
I also try to train Not using pre-trained weights only using own dataset.
It has training loss 1.4xx~ now.

Both training iteration are larger than 45000.
